src/frontend/src/instance/prom_store.rs

```diff
@@ -149,9 +149,7 @@ impl Instance {
                 .handle_remote_query(&ctx, catalog_name, schema_name, &table_name, query)
                 .await
                 .map_err(BoxedError::new)
-                .with_context(|_| error::ExecuteQuerySnafu {
-                    query: format!("{query:#?}"),
-                })?;
+                .with_context(|_| error::ExecuteQuerySnafu)?;
```

Comment:
I test locally a sensitive query: `COPY tbl FROM '/path/to/folder/xxx.parquet' WITH (FORMAT = 'parquet', SECRET_ACCESS_KEY=12345);`

It logs:

```
2024-04-25T19:19:37.144010Z ERROR on_query{protocol="mysql"}:mysql::do_query:do_query: frontend::instance: Failed to execute query: COPY tbl FROM /path/to/folder/xxx.parquet WITH (format = 'parquet', secret_access_key = '******') err=0: Table operation error, at src/frontend/src/instance.rs:273:14
1: Failed to build data source backend, at /Users/tison/GreptimeWorkspace/greptimedb/src/operator/src/statement/copy_table_from.rs:98:59
2: Failed to build backend, at src/common/datasource/src/object_store/fs.rs:25:10
3: Unexpected (permanent) at Builder::build => create root dir failed

2024-04-25T19:19:37.144742Z ERROR on_query{protocol="mysql"}: servers::mysql::writer: Failed to handle mysql query err=0: Failed to execute query, query: COPY tbl FROM '/path/to/folder/xxx.parquet' WITH (FORMAT = 'parquet', SECRET_ACCESS_KEY=12345), at /Users/tison/GreptimeWorkspace/greptimedb/src/servers/src/query_handler/sql.rs:89:22
1: Table operation error, at src/frontend/src/instance.rs:273:14
2: Failed to build data source backend, at /Users/tison/GreptimeWorkspace/greptimedb/src/operator/src/statement/copy_table_from.rs:98:59
3: Failed to build backend, at src/common/datasource/src/object_store/fs.rs:25:10
4: Unexpected (permanent) at Builder::build => create root dir failed
```

As you can see, the Display implementation can be redacted properly. But `ExecuteQuerySnafu` leaks the query.